common/clientCache.go

```diff
@@ -98,3 +100,11 @@ func (c *clientCacheImpl) GetClientForClientKey(clientKey string) (interface{},
 	c.clients[clientKey] = client
 	return client, nil
 }
+
+func (c *clientCacheImpl) GetAllClients() []string {
```

Comment:
This returns all keys. You can just return all clients here.

Reply:
Updated.